File: FRONTEND/src/app/shared/models/lote-recebido.ts
export interface LoteRecebido {
  id: number;
  numeroSerie: string;
  estetica: string;
  defeitoConstatadoId: number;
  classificacao: string;
  statusItem: string;
  resetado: string;
  capacidadeArmazenamento: number;
  possuiMarcasUso: boolean;
  loteTriagemId: number;
  triador: string;
  dataConferencia: string;
  statusConferencia: string;
  observacao: string;
}

export interface LoteRecebidoRequest {
  numeroSerie: string;
  estetica: string;
  defeitoConstatadoId: number | null;
  statusItem: string;
  resetado: string;
  capacidadeArmazenamento: number | null;
  possuiMarcasUso: boolean | null;
  loteTriagemId: number;
  triador: string;
  statusConferencia?: string;
  observacao?: string;
}

export interface DefeitoConstatado {
  id: number;
  descricao: string;
}

File: BACKEND/src/main/java/br/com/waps/nexus/domain/lote/recebido/LoteRecebido.java
package br.com.waps.nexus.domain.lote.recebido;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "tb_lote_recebido")
@Getter
@Setter
public class LoteRecebido {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // ── Conferência física ──────────────────────────
    private String numeroSerie;
    private String estetica;
    private Integer defeitoConstatadoId;
    private String classificacao;
    private String statusItem;
    private String resetado;

    // ── Lote (FK simples, mesmo padrão do Produto) ──
    private Integer loteTriagemId;

    // ── Controle ─────────────────────────────────────
    private String triador;

    @CreationTimestamp
    @Column(name = "data_conferencia", updatable = false)
    private LocalDateTime dataConferencia;

    private String statusConferencia;
    private String observacao;

    // ── Armazenamento ────────────────────────────────
    private Integer capacidadeArmazenamento;
    private Boolean possuiMarcasUso;
}
